state-chain/custom-rpc/src/lib.rs

```diff
@@ -791,29 +814,50 @@ where
 
 	fn cf_pool_price(
 		&self,
-		from_asset: RpcAsset,
-		to_asset: RpcAsset,
+		from: RpcAsset,
+		to: RpcAsset,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<Option<PoolPriceV1>> {
+		self.client
+			.runtime_api()
+			.cf_pool_price(self.unwrap_or_best(at), from.try_into()?, to.try_into()?)
+			.map_err(to_rpc_error)
+	}
+
+	fn cf_pool_price_v2(
+		&self,
+		base_asset: RpcAsset,
+		quote_asset: RpcAsset,
 		at: Option<state_chain_runtime::Hash>,
-	) -> RpcResult<Option<PoolPrice>> {
+	) -> RpcResult<PoolPriceV2> {
+		let base = base_asset.try_into()?;
+		let quote = quote_asset.try_into()?;
+		let hash = self.unwrap_or_best(at);
+		let sell_price = self
+			.client
+			.runtime_api()
+			.cf_pool_price(hash, quote, base)
+			.map_err(to_rpc_error)?;
 		self.client
```

Comment:
I think this would be cleaner, like this:

```
Ok(PoolPriceV2 {
    base_asset,
    quote_asset,
    sell_asset: self
			.client
			.runtime_api()
			.cf_pool_price(hash, quote, base)
			.map_err(to_rpc_error)?,
    buy_asset: self
			.client
			.runtime_api()
			.cf_pool_price(hash, base, quote)
			.map_err(to_rpc_error)?,
})
```